### PR TITLE
Generalise local and CI build experiences

### PR DESCRIPTION
**Pull request by RussKie**
The app can be build with "dotnet" without custom scripts. 
Packaging and publishing moved into MSBuild targets.

### CODE REVIEW

**Comment by RussKie** [comment]
@maraf, this is a result of few late-night sessions, could you please verify that I haven't missed anything? Thank you.

File: Directory.Build.props [hunk 1]
```diff
@@ -5,6 +5,8 @@
     <LangVersion>latest</LangVersion>
     <NoWarn>$(NoWarn);1573;1591;1712</NoWarn>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <SuppressNETCoreSdkPreviewMessage>true</SuppressNETCoreSdkPreviewMessage>
+    <PackageManagerUIRuntimeIdentifier>win-x86</PackageManagerUIRuntimeIdentifier>
```
Comment (RussKie) [inline Directory.Build.props:9]: Is there a reason we can't target `win-x64`?

Comment (maraf) [reply Directory.Build.props:9]: I wasn't able to blame anything and don't have any recollection 🤷‍♂️

File: appveyor.yml [hunk 2]
```diff
@@ -39,21 +44,62 @@ dotnet_csproj:
 #       build configuration       #
 #---------------------------------#
 
+install:
+- ps: |
+    # Install the required .NET SDK
+    Invoke-WebRequest "https://dot.net/v1/dotnet-install.ps1" -OutFile "./dotnet-install.ps1"
+    ./dotnet-install.ps1 -JsonFile global.json -InstallDir 'C:\Program Files\dotnet'
```
Comment (maraf) [inline appveyor.yml:51]: The global.json is missing, is that intended? Is it why the CI is failing?

Comment (RussKie) [reply appveyor.yml]: No, I copied this from gitextensions where we use global.json. I restored the original command you had in the script.

**Comment by RussKie** [comment]
> Locally I'm getting a lot of NU1604 errors
> 
> ```
> D:\Development\GitExtensions\gitextensions.pluginmanager\src\PackageManager.UI\PackageManager.UI.csproj error NU1604: Warning As Error: Project dependency Neptuo.Exceptions does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [D:\Development\GitExtens
> ions\gitextensions.pluginmanager\GitExtensions.PluginManager.sln]
> ```

Did you `git clean -xfd` all you pulled this change?

```powershell
PS D:\Development\gitextensions.pluginmanager> git clean -xdf

PS D:\Development\gitextensions.pluginmanager> dotnet restore
  Determining projects to restore...
  Restored D:\Development\gitextensions.pluginmanager\src\PackageManager\PackageManager.csproj (in 297 ms).
  Restored D:\Development\gitextensions.pluginmanager\test\PackageManager.Tests\PackageManager.Tests.csproj (in 451 ms).
  Restored D:\Development\gitextensions.pluginmanager\src\PackageManager.NuGet\PackageManager.NuGet.csproj (in 453 ms).
  Restored D:\Development\gitextensions.pluginmanager\src\PackageManager.Cli\PackageManager.Cli.csproj (in 453 ms).
  Restored D:\Development\gitextensions.pluginmanager\test\PackageManager.NuGet.Tests\PackageManager.NuGet.Tests.csproj (in 453 ms).
  Restored D:\Development\gitextensions.pluginmanager\src\PackageManager.UI\PackageManager.UI.csproj (in 466 ms).
  Restored D:\Development\gitextensions.pluginmanager\src\GitExtensions.PluginManager\GitExtensions.PluginManager.csproj (in 1.07 sec).

PS D:\Development\gitextensions.pluginmanager> dotnet build
  Determining projects to restore...
  All projects are up-to-date for restore.
  PackageManager -> D:\Development\gitextensions.pluginmanager\src\PackageManager\bin\Debug\net8.0-windows\PackageManager.dll
  PackageManager.NuGet -> D:\Development\gitextensions.pluginmanager\src\PackageManager.NuGet\bin\Debug\net8.0-windows\PackageManager.NuGet.dll
  PackageManager.Tests -> D:\Development\gitextensions.pluginmanager\test\PackageManager.Tests\bin\Debug\net8.0-windows\PackageManager.Tests.dll
  PackageManager.Cli -> D:\Development\gitextensions.pluginmanager\src\PackageManager.Cli\bin\Debug\net8.0-windows\PackageManager.Cli.dll
  PackageManager.NuGet.Tests -> D:\Development\gitextensions.pluginmanager\test\PackageManager.NuGet.Tests\bin\Debug\net8.0-windows\PackageManager.NuGet.Tests.dll
  PackageManager.UI -> D:\Development\gitextensions.pluginmanager\src\PackageManager.UI\bin\Debug\net8.0-windows\win-x86\PackageManager.UI.dll
  GitExtensions.PluginManager -> D:\Development\gitextensions.pluginmanager\src\GitExtensions.PluginManager\bin\Debug\net8.0-windows\GitExtensions.PluginManager.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:04.94
```

**Comment by maraf** [comment]
> Did you `git clean -xfd` after you pulled this change?

Yes, I did

```sh
PS D:\Development\GitExtensions\gitextensions.pluginmanager> git clean -xfd
Removing artifacts/
Removing src/GitExtensions.PluginManager/bin/
Removing src/GitExtensions.PluginManager/obj/
Removing src/PackageManager.Cli/bin/
Removing src/PackageManager.Cli/obj/
Removing src/PackageManager.NuGet/bin/
Removing src/PackageManager.NuGet/obj/
Removing src/PackageManager.UI/bin/
Removing src/PackageManager.UI/obj/
Removing src/PackageManager/bin/
Removing src/PackageManager/obj/
Removing test/PackageManager.NuGet.Tests/bin/
Removing test/PackageManager.NuGet.Tests/obj/
Removing test/PackageManager.Tests/bin/
Removing test/PackageManager.Tests/obj/
PS D:\Development\GitExtensions\gitextensions.pluginmanager> dotnet restore
  Determining projects to restore...
D:\Development\GitExtensions\gitextensions.pluginmanager\src\GitExtensions.PluginManager\GitExtensions.PluginManager.csproj : error NU1604:
 Warning As Error: Project dependency GitExtensions.Extensibility does not contain an inclusive lower bound. Include a lower bound in the d
ependency version to ensure consistent restore results. [D:\Development\GitExtensions\gitextensions.pluginmanager\GitExtensions.PluginManag
er.sln]
```

**Comment by RussKie** [comment]
Hmm... that's very strange.

One discrepancy I noticed, which doesn't appear to be related to this change directly. In the published nupkg there's a runtimeconfig.json, which is absent from nupkg published by all the recent CI runs...
image

Any thoughts on this?

**Comment by RussKie** [comment]
> > Did you `git clean -xfd` after you pulled this change?
> 
> Yes, I did. The thing happens when I did a fresh clone in a different location
> 
> ```shell
> PS D:\Development\GitExtensions\gitextensions.pluginmanager> git clean -xfd
> Removing artifacts/
> Removing src/GitExtensions.PluginManager/bin/
> Removing src/GitExtensions.PluginManager/obj/
> Removing src/PackageManager.Cli/bin/
> Removing src/PackageManager.Cli/obj/
> Removing src/PackageManager.NuGet/bin/
> Removing src/PackageManager.NuGet/obj/
> Removing src/PackageManager.UI/bin/
> Removing src/PackageManager.UI/obj/
> Removing src/PackageManager/bin/
> Removing src/PackageManager/obj/
> Removing test/PackageManager.NuGet.Tests/bin/
> Removing test/PackageManager.NuGet.Tests/obj/
> Removing test/PackageManager.Tests/bin/
> Removing test/PackageManager.Tests/obj/
> PS D:\Development\GitExtensions\gitextensions.pluginmanager> dotnet restore
>   Determining projects to restore...
> D:\Development\GitExtensions\gitextensions.pluginmanager\src\GitExtensions.PluginManager\GitExtensions.PluginManager.csproj : error NU1604:
>  Warning As Error: Project dependency GitExtensions.Extensibility does not contain an inclusive lower bound. Include a lower bound in the d
> ependency version to ensure consistent restore results. [D:\Development\GitExtensions\gitextensions.pluginmanager\GitExtensions.PluginManag
> er.sln]
> ```

I just tried it on a different workstation, and the same result as previously - no warnings:
```
PS D:\Development\gitextensions.pluginmanager> git clean -xdf
Removing .vs/
Removing src/GitExtensions.PluginManager/bin/
Removing src/GitExtensions.PluginManager/obj/
Removing src/PackageManager.Cli/bin/
Removing src/PackageManager.Cli/obj/
Removing src/PackageManager.NuGet/bin/
Removing src/PackageManager.NuGet/obj/
Removing src/PackageManager.UI/bin/
Removing src/PackageManager.UI/obj/
Removing src/PackageManager/bin/
Removing src/PackageManager/obj/
Removing test/PackageManager.NuGet.Tests/bin/
Removing test/PackageManager.NuGet.Tests/obj/
Removing test/PackageManager.Tests/bin/
Removing test/PackageManager.Tests/obj/

PS D:\Development\gitextensions.pluginmanager> dotnet restore
  Determining projects to restore...
  Restored D:\Development\gitextensions.pluginmanager\src\PackageManager\PackageManager.csproj (in 1.88 sec).
  Restored D:\Development\gitextensions.pluginmanager\src\GitExtensions.PluginManager\GitExtensions.PluginManager.csproj (in 2.04 sec).
  Restored D:\Development\gitextensions.pluginmanager\test\PackageManager.Tests\PackageManager.Tests.csproj (in 2.57 sec).
  Restored D:\Development\gitextensions.pluginmanager\src\PackageManager.NuGet\PackageManager.NuGet.csproj (in 6.72 sec).
  Restored D:\Development\gitextensions.pluginmanager\src\PackageManager.Cli\PackageManager.Cli.csproj (in 6.72 sec).
  Restored D:\Development\gitextensions.pluginmanager\test\PackageManager.NuGet.Tests\PackageManager.NuGet.Tests.csproj (in 6.72 sec).
  Restored D:\Development\gitextensions.pluginmanager\src\PackageManager.UI\PackageManager.UI.csproj (in 15.18 sec).
```

What's .NET SDK do you have installed? `dotnet --info`

I'm on:
```
.NET SDK:
 Version:           8.0.400-preview.0.24324.5
 Commit:            736a2616db
 Workload version:  8.0.400-manifests.d7126b9e
 MSBuild version:   17.11.0+1192b22fd
```

**Comment by maraf** [comment]
> What's .NET SDK do you have installed?

```
.NET SDK:
 Version:           8.0.303
 Commit:            29ab8e3268
 Workload version:  8.0.300-manifests.dc0d2b22
 MSBuild version:   17.10.4+10fbfbf2e
```

Previously had 8.0.202. Same results

**Comment by maraf** [comment]
> In the published nupkg there's a runtimeconfig.json, which is absent from nupkg published by all the recent CI runs...

I don't have it in my locally built package
image

**Review by maraf — APPROVED** [review]

**Comment by RussKie** [comment]
> > In the published nupkg there's a runtimeconfig.json, which is absent from nupkg published by all the recent CI runs...
> 
> I don't have it in my locally built package image

I tried building Git Extensions with the latest build, and Git Extensions failed:

```
Preparing build artifacts...
  ICSharpCode.TextEditor -> C:\projects\gitextensions\externals\ICSharpCode.TextEditor\Project\bin\Release\net8.0-windows\publish\
  GitExtensions -> C:\projects\gitextensions\artifacts\Release\publish\GitExtensions\
  Downloading 'GitExtensions.PluginManager.3.0.0.228.zip'.
  Zipping directory "C:\projects\gitextensions\artifacts\Release\publish\GitExtensions\" to "C:\projects\gitextensions\artifacts\Release\publish\GitExtensions-Portable-x64-4.3.0.17869-1cd999e35.zip".
  MSBuild version 17.10.4+10fbfbf2e for .NET Framework
C:\projects\gitextensions\setup\installer\Product.wxs(405): error LGHT0103: The system cannot find the file 'C:\projects\gitextensions\artifacts\Release\publish\\GitExtensions.PluginManager\Output\PackageManager\PackageManager.UI.runtimeconfig.json'. [C:\projects\gitextensions\setup\installer\Setup.wixproj] [C:\projects\gitextensions\src\app\GitExtensions\GitExtensions.csproj]
C:\projects\gitextensions\src\app\GitExtensions\Project.Publish.targets(341,5): error MSB3073: The command "..." exited with code 1. [C:\projects\gitextensions\src\app\GitExtensions\GitExtensions.csproj]
Command exited with code 1
```

**Pull request merged**